src/module.ts

```diff
@@ -139,7 +139,13 @@ ${providers.map(p => `  ['${p.name}']: { provider: ${p.importName}, defaults: ${
         imageOptions.provider = options.provider = resolvedProvider
         options[resolvedProvider] = options[resolvedProvider] || {}
 
-        const p = await resolveProvider(nuxt, resolvedProvider, options[resolvedProvider])
+        const p = await resolveProvider(
+          nuxt,
+          resolvedProvider,
+          BuiltInProviders.includes(resolvedProvider)
+            ? { provider: resolvedProvider, options: options[resolvedProvider] }
+            : options[resolvedProvider]
```

Comment:
Was this change necessary? (I think it would be nice to keep this arg unified with `{ options }` interface always.)

Reply:
It's necessary.
The `resolveProviders` that take `ModuleOptions` as arguments also do this processing, so they need to do the same for `ModuleOptions` in the `nitro:init` hook.
https://github.com/nuxt/image/blob/v1.0.0-rc.3/src/provider.ts#L67-L71

Reply:
Considering this logic running within `if (!options.provider || options.provider === 'ipx' || options.provider === 'ipxStatic')`, possible values for `resolvedProvider` are `ipx`, `ipxStatic` and `none`, all are built-in so i think this line would not also reach...